contracts/potlock/src/potlock_storage.rs

```diff
@@ -0,0 +1,111 @@
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub type PotlockId = usize;
+pub type ProjectId = usize;
+
+#[derive(TypeAbi, TopEncode, TopDecode, PartialEq, Eq, Debug, NestedEncode, NestedDecode)]
+pub enum Status {
+    Inactive,
+    Active,
+}
+
+#[derive(TypeAbi, NestedEncode, NestedDecode, PartialEq, Debug, TopEncode, TopDecode)]
+pub struct Pot<M: ManagedTypeApi> {
+    pub potlock_id: PotlockId,
+    pub token_identifier: TokenIdentifier<M>,
+    pub fee: BigUint<M>,
+    pub name: ManagedBuffer<M>,
+    pub description: ManagedBuffer<M>,
+    pub status: Status,
+}
+
+impl<M: ManagedTypeApi> Pot<M> {
+    pub fn new(
+        potlock_id: PotlockId,
+        name: ManagedBuffer<M>,
+        description: ManagedBuffer<M>,
+    ) -> Self {
+        Pot {
+            potlock_id,
+            token_identifier: TokenIdentifier::from(ManagedBuffer::default()),
+            fee: BigUint::default(),
+            name,
+            description,
+            status: Status::Inactive,
+        }
+    }
+}
+
+#[derive(TypeAbi, NestedEncode, NestedDecode, PartialEq, Debug, TopEncode, TopDecode)]
+pub struct Project<M: ManagedTypeApi> {
+    pub project_id: ProjectId,
+    pub potlock_id: PotlockId,
+    pub name: ManagedBuffer<M>,
+    pub description: ManagedBuffer<M>,
+    pub owner: ManagedAddress<M>,
+    pub status: Status,
+}
+
+impl<M: ManagedTypeApi> Project<M> {
+    pub fn new(
+        project_id: ProjectId,
+        potlock_id: PotlockId,
+        name: ManagedBuffer<M>,
+        description: ManagedBuffer<M>,
+        owner: ManagedAddress<M>,
+    ) -> Self {
+        Project {
+            project_id,
+            potlock_id,
+            name,
+            description,
+            owner,
+            status: Status::Inactive,
+        }
+    }
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, PartialEq, Debug)]
+pub struct UserDonations<M: ManagedTypeApi> {
+    pub user: ManagedAddress<M>,
+    pub donations: EsdtTokenPayment<M>,
+}
+
+#[multiversx_sc::module]
+pub trait PotlockStorage {
+    #[view(getFeeTokenIdentifier)]
+    #[storage_mapper("feeTokenIdentifier")]
+    fn fee_token_identifier(&self) -> SingleValueMapper<TokenIdentifier>;
+
+    #[view(getFeeAmount)]
+    #[storage_mapper("feeAmount")]
+    fn fee_amount(&self) -> SingleValueMapper<BigUint>;
+
+    #[view(getPotlocks)]
+    #[storage_mapper("potlocks")]
+    fn potlocks(&self) -> VecMapper<Pot<Self::Api>>;
```

Comment:
UnorderedSetMapper will reorder the set.
If the set contains Pots at indexes: 1, 2, 3, 4 and Pot #2 is removed, then the set will be reordered to 1, 4, 3.
When we try to retrieve Pot #2, we will get Pot #4 instead of returning an error with "Pot doesn't exist".